### DISCUSSION TITLE
Flutter Rust Bridge State When Flutter App Lifecycle is Detached

### DISCUSSION THREAD

**Discussion by szczygielmr** [fzyzcjy/flutter_rust_bridge#2361] [Q&A] [answered]
What happens to Flutter Rust Bridge (FRB) when a Flutter app's lifecycle becomes detached?  Is the FRB still executing in the background?  If so when the Flutter app is in a resumed state, how can I get a handle to the executing FRB?

**Comment by fzyzcjy** [comment]
May I know what do you mean by "executing"?

frb is just a "bridge" to allow Rust and Dart/Flutter communicate easily. Thus the answer is same as if you have some vanilla Rust (or C or C++) code and vanilla Dart code.

For example, suppose the whole app process is killed, then both Dart and Rust code will stop.

**Reply by szczygielmr** [reply]
When I say "executing", in rust we are spawning ractor actors, https://github.com/slawlor/ractor, that are executing various work in the background.  I would like to keep these background processes running when the Flutter application is in a detached state.  

The rust actor logic still seems to be executing while the Flutter app is in a detached state.  In rust, I verified this by writing a heartbeat message to a file while in this state.  When the Flutter application is resumed, is there a way to reclaim the handle or "rebuild the bridge" to the running rust code?

**Reply by fzyzcjy** [reply, marked as answer]
>  I would like to keep these background processes running when the Flutter application is in a detached state.

So, do you want to make it run for e.g. when user completely exit your app? Then I guess you need to find some doc about underlying platform, or better, some Rust/Dart wrappers. IIRC e.g. Android allows some background service to be long living even if user exits the app, as long as you follow their doc to setup it.

> When the Flutter application is resumed, is there a way to reclaim the handle or "rebuild the bridge" to the running rust code?

If the Flutter app is not exited, then the state in Flutter is still there, thus just do everything normally imho.
